initialization/initialization.go

```diff
@@ -234,16 +234,28 @@ func (init *Initializer) Initialize(ctx context.Context) error {
 	}
 	defer init.mtx.Unlock()
 
-	numLabels := uint64(init.opts.NumUnits) * uint64(init.cfg.LabelsPerUnit)
-	fileNumLabels := numLabels / uint64(init.opts.NumFiles)
+	init.logger.Info("initialization: datadir: %v, number of units: %v, max file size: %v, number of labels per unit: %v, number of bits per label: %v",
+		init.opts.DataDir, init.opts.NumUnits, init.opts.MaxFileSize, init.cfg.LabelsPerUnit, init.cfg.LabelsPerUnit)
```

Comment:
How about using structured logs here?
```suggestion
	init.logger.With().Info("initialization", log.String("datadir", init.opts.DataDir), (...))
```

Reply:
For this we would need to add a hard dependency on `zap`, which I'm not against btw. Instead of `shared.Logger` `WithLogger` should take a `zap.Logger`.

I don't like the `shared.Logger` interface that we currently have as abstraction. It needs an adapter so we can use `zap` and the adapter uses the `SugaredLogger`, which is slower and uses more memory than the non-`Sugared` logger.

The only reason I see for the abstraction is for testing, where we can just use `zaptest.NewLogger`.

Reply:
The current logger doesn't support it, so I'd say that it's outside of the scope of this PR.

Reply:
I created an issue for this to fix it in a future PR: https://github.com/spacemeshos/post/issues/95